File: org.kevoree.modeling.microframework/src/test/java/org/kevoree/modeling/microframework/test/poc/impl/CloudViewImpl.java
package org.kevoree.modeling.microframework.test.poc.impl;

import org.kevoree.modeling.api.Callback;
import org.kevoree.modeling.api.KDimension;
import org.kevoree.modeling.api.KObject;
import org.kevoree.modeling.api.abs.AbstractKView;
import org.kevoree.modeling.api.data.DataCache;
import org.kevoree.modeling.api.meta.MetaClass;
import org.kevoree.modeling.microframework.test.poc.CloudView;
import org.kevoree.modeling.microframework.test.poc.Element;
import org.kevoree.modeling.microframework.test.poc.Node;

import java.util.List;

/**
 * Created by duke on 10/9/14.
 */
public class CloudViewImpl extends AbstractKView implements CloudView {

    protected CloudViewImpl(long now, KDimension dimension, DataCache cache) {
        super(now, dimension, cache);
    }

    @Override
    public KObject createFQN(String metaClassName) {
        return null;
    }

    @Override
    public KObject create(MetaClass clazz) {
        switch (clazz.index()) {
            case 0:
                return internal_create(new NodeImpl(this, CloudView.METACLASSES.org_kevoree_modeling_microframework_test_poc_Node, now(), dimension(), null));
            case 1:
                return internal_create(new ElementImpl(this, METACLASSES.org_kevoree_modeling_microframework_test_poc_Element, now(), dimension(), null));
            default:
                return null;
        }
    }

    private KObject internal_create(KObject obj) {
        //TODO put into cache, iniate values cache
        return obj;
    }

    @Override
    public void root(KObject elem, Callback<Boolean> callback) {

    }

    @Override
    public void select(String query, Callback<List<KObject>> callback) {

    }

    @Override
    public void lookup(String path, Callback<KObject> callback) {

    }

    @Override
    public void stream(String query, Callback<KObject> callback) {

    }

    @Override
    public MetaClass[] metaClasses() {
        return CloudView.METACLASSES.values();
    }

    @Override
    public Node createNode() {
        return (Node) create(METACLASSES.org_kevoree_modeling_microframework_test_poc_Node);
    }

    @Override
    public Element createElement() {
        return (Element) create(METACLASSES.org_kevoree_modeling_microframework_test_poc_Element);
    }
}

File: org.kevoree.modeling.microframework.typescript/src/test/ts/org/kevoree/modeling/microframework/test/cloud/impl/CloudViewImpl.ts
///<reference path="../../../../api/KDimension.ts"/>
///<reference path="../../../../api/KObject.ts"/>
///<reference path="../../../../api/abs/AbstractKView.ts"/>
///<reference path="../../../../api/meta/MetaClass.ts"/>
///<reference path="../../../../api/time/TimeTree.ts"/>
///<reference path="../CloudView.ts"/>
///<reference path="../Element.ts"/>
///<reference path="../Node.ts"/>

class CloudViewImpl extends AbstractKView implements CloudView {

  constructor(now: number, dimension: KDimension) {
    super(now, dimension);
  }

  public internalCreate(clazz: MetaClass, timeTree: TimeTree, key: number): KObject {
    if (clazz == null) {
      return null;
    }
    switch (clazz.index()) {
      case 0: 
      return manageCache(new NodeImpl(this, METACLASSES.ORG_KEVOREE_MODELING_MICROFRAMEWORK_TEST_CLOUD_NODE, key, now(), dimension(), timeTree));
      case 1: 
      return manageCache(new ElementImpl(this, METACLASSES.ORG_KEVOREE_MODELING_MICROFRAMEWORK_TEST_CLOUD_ELEMENT, key, now(), dimension(), timeTree));
      default: 
      return null;
    }
  }

  public metaClasses(): MetaClass[] {
    return CloudView.METACLASSES.values();
  }

  public createNode(): Node {
    return <Node>create(METACLASSES.ORG_KEVOREE_MODELING_MICROFRAMEWORK_TEST_CLOUD_NODE);
  }

  public createElement(): Element {
    return <Element>create(METACLASSES.ORG_KEVOREE_MODELING_MICROFRAMEWORK_TEST_CLOUD_ELEMENT);
  }

}

